code/unclassified/src/running_median_heap/running_median_heap.cpp

```diff
@@ -0,0 +1,216 @@
+* If the input size is even, we pick average of middle two elements in 
+* sorted stream.
+*
+* Time Complexity: If we omit the way how stream was read, 
+* complexity of median finding is O(N log N), as we need to read the stream, 
+* and due to heap insertions/deletions.
+*/
+
+#include <iostream>
+#include <functional>
+
+// Utility functions
+float average(int a, int b)
+{
+    return (a + b) / 2.0;
+}
+
+/*
+* Signum function
+* = 0  if a == b  - heaps are balanced
+* = -1 if a < b   - left contains less elements than right
+* = 1  if a > b   - left contains more elements than right
+*/
+int signum(int a, int b)
+{
+    if (a == b)
+        return 0;
+
+    return a < b ? -1 : 1;
+}
+
+class Heap 
+{ 
+    float A[1024]; // array of elements in heap with capacity of 1024 elements.
+    int heapSize_; // Current number of elements in min heap
+    std::function<bool(float, float)> comp;  //comparator function for min/max heap
+
+public: 
+    // Constructor 
+    Heap(std::function<bool(float, float)> c) : comp(c){heapSize_ = 0;} 
+ 
+    // Returns number of elements in heap
+    int getCount() {return heapSize_;}
+  
+    // to heapify a subtree with the root at given index 
+    void heapify(int idx) 
+    { 
+        int l = left(idx); 
+        int r = right(idx); 
+        int smallest = idx; 
+        if (l < heapSize_ && comp(A[l], A[idx])) 
+            smallest = l; 
+        if (r < heapSize_ && comp(A[r], A[smallest])) 
+            smallest = r; 
+        if (smallest != idx) 
+        { 
+            std::swap(A[idx], A[smallest]); 
+            heapify(smallest); 
+        } 
+    } 
+
+    int parent(int idx) {return (idx-1)/2;} 
+
+    // to get index of left child of node at index i 
+    int left(int idx) {return (2*idx+ 1);} 
+
+    // to get index of right child of node at index i 
+    int right(int idx) {return (2*idx + 2);} 
+
+    // to extract the root which is the min/max element 
+    float extractTop() 
+    {  
+        if (heapSize_ == 1) 
+        { 
+            heapSize_--; 
+            return A[0]; 
+        } 
+
+        // Store the return value, and remove it from heap 
+        float root = A[0]; 
+        A[0] = A[heapSize_-1]; 
+        heapSize_--; 
+        heapify(0); 
+      
+        return root; 
+    } 
+
+    // Returns the element (key at root) from heap 
+    float getTop() {return A[0];} 
+  
+    // Inserts a new element 
+    void insert(int currElement) 
+    { 
+        // First insert the new key at the end 
+        heapSize_++; 
+        int idx = heapSize_ - 1; 
+        A[idx] = currElement; 
+
+        // Fix the heap property if it is violated 
+        while (idx != 0 && comp(A[idx], A[parent(idx)])) 
+        { 
+           std::swap(A[idx], A[parent(idx)]); 
+           idx = parent(idx); 
+        } 
+    } 
+}; 
+
+// Function implementing algorithm to find median so far.
+float getMedian(float currElement, float &median, Heap &left, Heap &right) 
```

Comment:
Don't modify the value of median passed in as a parameter, just return it. It's better coding practice.